PsfLauncher/main.cpp

```diff
@@ -268,43 +277,233 @@ void LaunchMonitorInBackground(std::filesystem::path packageRoot, const wchar_t
         }
         else
         {
-            //Log("error starting monitor using SellExecuteEx also. Error=0x%x\n", ::GetLastError());
+            std::wostringstream ss;
+            ss << L"error starting monitor using SellExecuteEx";
+            auto err = ::GetLastError();
+            ErrorInformation error(ss.str(), err, executable);
+
+            return error;
         }
     }
     else
     {
-        STARTUPINFO startupInfo = { sizeof(startupInfo) };
-        startupInfo.dwFlags = STARTF_USESHOWWINDOW;
-        startupInfo.wShowWindow = static_cast<WORD>(1);
+        std::wstring cmdarg = cmd + L" " + arguments;
 
-        PROCESS_INFORMATION processInfo;
+        ExecutionInformation execInfo;
+        execInfo.ApplicationName = nullptr;
+        execInfo.CommandLine = cmdarg;
 
-        std::wstring cmdarg = cmd + L" " + arguments;
+        auto currentDirectory = (packageRoot / dirStr);
+        execInfo.CurrentDirectory = currentDirectory.c_str();
+        ErrorInformation error = StartProcess(execInfo, cmdShow, false);
+        error.AddExeName(executable);
+
+        return error;
+    }
+
+    ErrorInformation noError;
+    return noError;
+}
 
-        if (!::CreateProcessW(
-            nullptr, //quotedapp.data(),
-            (wchar_t *)cmdarg.c_str(),
-            nullptr, nullptr, // Process/ThreadAttributes
-            true, // InheritHandles
-            0, // CreationFlags
-            nullptr, // Environment
-            nullptr,
-            &startupInfo,
-            &processInfo))
+ErrorInformation StartProcess(ExecutionInformation execInfo, int cmdShow, bool runInAppContainer)
+{
+    STARTUPINFOEXW StartupInfoEx = { 0 };
+
+    StartupInfoEx.StartupInfo.cb = sizeof(StartupInfoEx);
+    StartupInfoEx.StartupInfo.dwFlags = STARTF_USESHOWWINDOW;
+    StartupInfoEx.StartupInfo.wShowWindow = static_cast<WORD>(cmdShow);
+
+    if (runInAppContainer)
+    {
+        SIZE_T AttributeListSize;
+        InitializeProcThreadAttributeList(NULL, 1, 0, &AttributeListSize);
+        StartupInfoEx.lpAttributeList = (LPPROC_THREAD_ATTRIBUTE_LIST)HeapAlloc(
+            GetProcessHeap(),
+            0,
+            AttributeListSize
+        );
+
+        if (InitializeProcThreadAttributeList(StartupInfoEx.lpAttributeList,
+            1,
+            0,
+            &AttributeListSize) == FALSE)
         {
-            if (::GetLastError() == ERROR_ELEVATION_REQUIRED)
-                Log("error starting monitor using CreateProcessW. You must specify 'monitor/asadmin' in config.json\n");
-            else
-                Log("error starting monitor using CreateProcessW. Error=0x%x\n", ::GetLastError());
+            auto err{ ::GetLastError() };
+            std::wostringstream ss;

```

Comment:
>wostringstream [](start = 17, length = 14)

no need for this

Reply:
Addressed in another comment.  Changed to std::wstring.